# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsIProcessSwitchRequestor.idl
# Commit: 5686e9cc48b2
# Full Hash: 5686e9cc48b26a8ec550f94091d5b8b4ed840f7d
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2019-10-22 09:46:06
# Regressor Bug: 1579992
# File Overlap Count: 1
# Description:
#   Bug 1579992 - Load pages into new webCOOP+COEP process type r=nika
#   
#   * This patch makes pages with the `OPENER_POLICY_SAME_ORIGIN_EMBEDDER_POLICY_REQUIRE_CORP` policy load into a special `webCOOP+COEP={pageOrigin}` remote type.
#   * Adds `E10SUtils.WEB_REMOTE_COOP_COEP_TYPE_PREFIX="webCOOP+COEP="`
#   * When a COOP process switch occurs and the target page doesn't have this policy, we pass a `preferredRemoteType="web"` into `E10SUtils.getRemoteTypeForPrincipal` ensuring that we correctly get a different `remoteType`
# ==============================================================================

diff -r fd0d2f340380 -r 5686e9cc48b2 netwerk/base/nsIProcessSwitchRequestor.idl
--- a/netwerk/base/nsIProcessSwitchRequestor.idl	Mon Oct 21 16:34:43 2019 +0000
+++ b/netwerk/base/nsIProcessSwitchRequestor.idl	Mon Oct 21 16:56:00 2019 +0000
@@ -42,4 +42,11 @@
    * @throws NS_ERROR_NOT_AVAILABLE if we don't have a responseHead
    */
   [must_use] boolean hasCrossOriginOpenerPolicyMismatch();
+
+  /**
+   * Returns a cached CrossOriginOpenerPolicy that is computed just before we
+   * determine if there is a policy mismatch.
+   * @throws NS_ERROR_NOT_AVAILABLE if it has not been computed yet
+   */
+  [must_use] readonly attribute nsILoadInfo_CrossOriginOpenerPolicy crossOriginOpenerPolicy;
 };